### PR TITLE
refactor: Renames AmplifyAuthError to AuthError

### PR DESCRIPTION
**Pull request by royjit**
BREAKING CHANGE: Auth category error type is changed to AuthError, current implementation that make use of AmplifyAuthError
will break with build time error.

Also deleted AuthError in storage category.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthService.swift [hunk 1]
```diff
@@ -81,31 +81,8 @@ public class AWSAuthService: AWSAuthServiceBehavior {
...
-                                              "The user needs be authenticated to make this request",
-                                              error)
-        case .notAuthorized(let message):
-            return AuthError.notAuthorized(message,
-                                           "The user does not have the correct permissions to make this request",
-                                           error)
-        default:
-            return AuthError.unknown(error.message)
-        }
+        // TODO: Use main auth error from Auth plugin
```
Comment (lawmicha) [inline AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthService.swift:84]: what does this mean?

Comment (royjit) [reply AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthService.swift:84]: My next PR will handle this, basically to use the error coming from Auth plugin. We will not need this mapping method since we will be removing AWSMobileClient from AWSPluginsCore

Comment (lawmicha) [reply AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthService.swift:84]: these look like they were specific to when using storage, need to verify Storage Integration tests negative paths

Comment (lawmicha) [reply AmplifyPlugins/Core/AWSPluginsCore/Auth/AWSAuthService.swift:84]: https://github.com/aws-amplify/amplify-ios/issues/435
